src/logging.cpp

```diff
@@ -236,11 +237,15 @@ namespace BCLog {
     }
 }
 
-void BCLog::Logger::LogPrintStr(const std::string& str)
+void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& logging_function, const std::string& source_file, const int source_line)
 {
     StdLockGuard scoped_lock(m_cs);
     std::string str_prefixed = LogEscapeMessage(str);
 
+    if (m_log_functionnames && m_started_new_line) {
+        str_prefixed.insert(0, "[" + source_file + ":" + ToString(source_line) + "] [" + logging_function + "] ");
+    }
```

Comment:
If I'd written it, I would have preferred `[foo.cpp:123:Function()]` vs `[foo.cpp:123] [Function]`

Reply:
Slight preference towards this style as well.

Reply:
I tried with that, but I find the current version easier to parse.

As-is:

```
2021-01-15T10:00:03Z [init.cpp:1568] [AppInitMain] * Using 440.0 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2021-01-15T10:00:03Z [noui.cpp:56] [noui_InitMessage] init message: Loading block index...                                                       
2021-01-15T10:00:03Z [validation.cpp:5171] [InitializeChainstate] Switching active chainstate to Chainstate [ibd] @ height -1 (null)
2021-01-15T10:00:03Z [dbwrapper.cpp:137] [CDBWrapper] Opening LevelDB in /root/.bitcoin/regtest/blocks/index
2021-01-15T10:00:03Z [dbwrapper.cpp:141] [CDBWrapper] Opened LevelDB successfully                                                         
```

Alternative version suggested by @ajtowns:

```
2021-01-15T10:00:03Z [init.cpp:1568:AppInitMain] * Using 440.0 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2021-01-15T10:00:03Z [noui.cpp:56:noui_InitMessage] init message: Loading block index...                                                       
2021-01-15T10:00:03Z [validation.cpp:5171:InitializeChainstate] Switching active chainstate to Chainstate [ibd] @ height -1 (null)
2021-01-15T10:00:03Z [dbwrapper.cpp:137:CDBWrapper] Opening LevelDB in /root/.bitcoin/regtest/blocks/index
2021-01-15T10:00:03Z [dbwrapper.cpp:141:CDBWrapper] Opened LevelDB successfully                                                         
```

I find the long `[…]` blocks a bit heavy to parse TBH :)

Keeping as-is :)